eras/conway/test-suite/src/Test/Cardano/Ledger/Conway/Serialisation/Roundtrip.hs

```diff
@@ -23,5 +32,25 @@ allprops =
   testGroup
     (show $ typeRep (Proxy @e))
     [ testProperty "ConwayGenesis" $
-        roundTripCborExpectation @(ConwayGenesis (EraCrypto e)) (eraProtVerHigh @e)
+        roundTripCborExpectation @(ConwayGenesis (EraCrypto e)) (eraProtVerHigh @e),
+      testProperty "conway/TxOut with pointer addresses" ptrAddrTxOutNotDeserialized,
+      testProperty "conway/TxOut without pointer addresses" nonPtrAddrTxOutDeserialized
     ]
+
+ptrAddrTxOutNotDeserialized :: Property
+ptrAddrTxOutNotDeserialized =
+  forAll (genPtrAddress @(EraCrypto Conway)) $ \ptrAddr txOut ->
+    do
+      let ptrAddrTxOut = txOut & addrTxOutL .~ ptrAddr
+      let messageMatch s = "Pointer addresses not supported starting with version 9" `isInfixOf` s
+      property $
+        case (roundTrip (natVersion @9) (cborTrip @(BabbageTxOut Conway)) ptrAddrTxOut) of
+          Left (RoundTripFailure _ _ _ _ _ _ (Just e)) -> messageMatch (show e)
+          _ -> False
+
+nonPtrAddrTxOutDeserialized :: Property
+nonPtrAddrTxOutDeserialized =
+  forAll (genNonPtrAddress @(EraCrypto Conway)) $ \nonPtrAddr txOut ->
+    do
+      let nonPtrAddrTxOut = txOut & addrTxOutL .~ nonPtrAddr
+      property $ isRight $ (roundTrip (natVersion @9) (cborTrip @(BabbageTxOut Conway)) nonPtrAddrTxOut)
```

Comment:
Redundant parenthesis, also isRight will not tell us anything why this test fails, if it does:
```suggestion
      counterexample ("Unexpected decoder failure: " ++ show err) $
      case roundTrip (natVersion @9) (cborTrip @(BabbageTxOut era)) nonPtrAddrTxOut of
        Left err ->  False
        Right  -> True
```